Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs

```diff
@@ -0,0 +1,292 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.UI;
+using IBM.Watson.SpeechToText.V1;
+using IBM.Cloud.SDK;
+using IBM.Cloud.SDK.Authentication;
+using IBM.Cloud.SDK.Authentication.Iam;
+using IBM.Cloud.SDK.Utilities;
+using IBM.Cloud.SDK.DataTypes;
+using TMPro;
+
+namespace IBM.Watsson.Examples
+{
+    public class CaptionGenerator : MonoBehaviour
+    {
+        [SerializeField] private TMP_Text ResultsField;
+        [SerializeField] private Button _stopButton;
+        private string _serviceUrl;
+        private string _iamApikey;
+        private string _recognizeModel;
+        private int _recordingRoutine = 0;
+        private string _microphoneID = null;
+        private AudioClip _recording = null;
+        private int _recordingBufferSize = 1;
+        private int _recordingHZ = 22050;
+        private SpeechToTextService _service;
+        private string accumulatedText = "";
+        
+        void Start()
+        {
+            _stopButton.onClick.AddListener(StopService);
+            LoadKeysFromEnvFile();
+            Debug.Log("API Key: " + _iamApikey);
+            Debug.Log("URL: " + _serviceUrl);
+            LogSystem.InstallDefaultReactors();
+            Runnable.Run(CreateService());
+        }
+
+        private void LoadKeysFromEnvFile()
+        {
+            string pathToEnvFile = Application.dataPath + "/../ibm-credentials.env";
```

Comment:
file storage and downloading is not possible in this way.
it simply won’t make it into the build for mobile devices.
I'll rewrite this.

file storage and downloading is not possible in this way.
it simply won’t make it into the build for mobile devices.
I'll rewrite this.

the file with the keys will need to be renamed to "caption-ibm-credentials.txt" and placed in the path "{Project_Root}\Assets\Resources"